api/OffsetCommit/request.go

```diff
@@ -4,15 +4,21 @@ import (
 	"github.com/mkocikowski/libkafka/api"
 )
 
-func NewRequest(group, topic string, partition int32, offset, retentionMs int64) *api.Request {
-	p := Partition{
-		PartitionIndex:   partition,
-		CommitedOffset:   offset,
-		CommitedMetadata: "",
+// NewRequest constructs api.Request with ApiKey "OffsetCommit". Method supports
+// a map of offsets for multiple partitions
+func NewRequest(group, topic string, offsets map[int32]int64, retentionMs int64) *api.Request {
+	offsetsMap := make([]Partition, len(offsets))
+	i := 0
```

Comment:
variable "i" does nothing

Reply:
What do you mean by does nothing? Variable `i` does the right job to fulfil the offsets map.